### PR TITLE
removing docker from external ci pipelines

### PR DESCRIPTION
**Pull request by abhimeda**
co-authored by Joseph Macaranas, Alex Xu, and Daniel Su.

### CODE REVIEW

File: .azuredevops/components/hipSPARSELt.yml [hunk 1]
```diff
@@ -38,17 +57,32 @@ jobs:
...
     parameters:
       extraBuildFlags: >-
         -DCMAKE_BUILD_TYPE=Release
-        -DCMAKE_CXX_COMPILER=/opt/rocm/bin/amdclang++
-        -DCMAKE_C_COMPILER=/opt/rocm/bin/amdclang
+        -DCMAKE_CXX_COMPILER=$(Agent.BuildDirectory)/rocm/llvm/bin/amdclang++
+        -DCMAKE_C_COMPILER=$(Agent.BuildDirectory)/rocm/llvm/bin/amdclang
         -DAMDGPU_TARGETS=all
         -DTensile_LOGIC=
         -DTensile_CPU_THREADS=
```
Comment (alexxu-amd) [inline .azuredevops/components/hipSPARSELt.yml:81]: These flags have no value. Maybe we can delete them to make it cleaner

File: .azuredevops/components/hipSPARSELt.yml [hunk 2]
```diff
@@ -38,17 +57,32 @@ jobs:
...
         -DTensile_CODE_OBJECT_VERSION=default
         -DTensile_LIBRARY_FORMAT=msgpack
-        -DCMAKE_PREFIX_PATH="/opt/rocm"
+        -DCMAKE_PREFIX_PATH="$(Agent.BuildDirectory)/rocm"
+        -DTENSILE_ROCM_ASSEMBLER_PATH=$(Agent.BuildDirectory)/rocm/llvm/bin/amdclang++
```
Comment (alexxu-amd) [inline .azuredevops/components/hipSPARSELt.yml:85]: No need DTENSILE_ROCM_ASSEMBLER_PATH anymore as it is defined in the environment

**Review by alexxu-amd — CHANGES_REQUESTED** [review]

File: .azuredevops/components/hipSPARSELt.yml (hunk 2 above)
Comment (abhimeda) [reply .azuredevops/components/hipSPARSELt.yml]: done

File: .azuredevops/components/hipSPARSELt.yml (hunk 1 above)
Comment (abhimeda) [reply .azuredevops/components/hipSPARSELt.yml:79]: done

Comment (alexxu-amd) [reply .azuredevops/components/hipSPARSELt.yml:81]: Please keep the compiler flags, I was referring to DTensile_LOGIC and -DTensile_CPU_THREADS

Comment (abhimeda) [reply .azuredevops/components/hipSPARSELt.yml:81]: The build script for this at https://github.com/ROCm/rocm-build/blob/rocm-rel-6.1/build/build_hipsparselt.sh are specifically written in that way. I think we should keep them as is. I will try to remove them as well just to test out though.

**Comment by alexxu-amd** [comment]
The branch has some merge conflicts to resolve. Also double check with Joseph with the pool settings before you merge. The rest looks good!

File: .azuredevops/components/rocFFT.yml [hunk 3]
```diff
@@ -48,4 +71,5 @@ jobs:
         -DBUILD_CLIENTS_BENCHMARKS=OFF
         -DBUILD_CLIENTS_SAMPLES=OFF
         -GNinja
+
```
Comment (danielsu-amd) [inline .azuredevops/components/rocFFT.yml:74]: An extra new line here

**Review by danielsu-amd — CHANGES_REQUESTED** [review]

File: .azuredevops/components/rocFFT.yml (hunk 3 above)
Comment (abhimeda) [reply .azuredevops/components/rocFFT.yml]: fixed

**Comment by amd-jmacaran** [comment]
> The branch has some merge conflicts to resolve. Also double check with Joseph with the pool settings before you merge. The rest looks good!

hipFFT should be fine on base pool. Must have missed it being on medium before.

**Comment by abhimeda** [comment]
> > The branch has some merge conflicts to resolve. Also double check with Joseph with the pool settings before you merge. The rest looks good!
> 
> hipFFT should be fine on base pool. Must have missed it being on medium before.

would you like me to change rocFFT to base pool?

**Review by danielsu-amd — APPROVED** [review]

**Comment by amd-jmacaran** [comment]
> > > The branch has some merge conflicts to resolve. Also double check with Joseph with the pool settings before you merge. The rest looks good!
> > 
> > 
> > hipFFT should be fine on base pool. Must have missed it being on medium before.
> 
> would you like me to change rocFFT to base pool?

No

**Review by alexxu-amd — APPROVED** [review]

**Pull request merged**